src/Components/WebAssembly/testassets/ThreadingApp/Pages/FetchData.razor

```diff
@@ -0,0 +1,90 @@
+﻿@using Microsoft.AspNetCore.Components.WebAssembly.Http;
+@page "/fetchdata"
+@page "/fetchdata/{StartDate:datetime}"
+@inject HttpClient Http
+
+
+<h1>Weather forecast</h1>
+
+<p>This component demonstrates fetching data from the server.</p>
+
+@if (forecasts == null)
+{
+    <p><em>Loading...</em></p>
+}
+else
+{
+    <table class='table'>
+        <thead>
+            <tr>
+                <th>Date</th>
+                <th>Temp. (C)</th>
+                <th>Temp. (F)</th>
+                <th>Summary</th>
+            </tr>
+        </thead>
+        <tbody>
+            @foreach (var forecast in forecasts)
+            {
+                <tr>
+                    <td>@forecast.DateFormatted</td>
+                    <td>@forecast.TemperatureC</td>
+                    <td>@forecast.TemperatureF</td>
+                    <td>@forecast.Summary</td>
+                </tr>
+            }
+        </tbody>
+    </table>
+    <p>
+        <a href="fetchdata/@startDate.AddDays(-5).ToString("yyyy-MM-dd")" class="btn btn-secondary float-left">
+            ◀ Previous
+        </a>
+        <a href="fetchdata/@startDate.AddDays(5).ToString("yyyy-MM-dd")" class="btn btn-secondary float-right">
+            Next ▶
+        </a>
+    </p>
+}
+
+@code {
+    [Parameter] public DateTime? StartDate { get; set; }
+
+    WeatherForecast[] forecasts;
+    DateTime startDate;
+
+    protected override async Task OnParametersSetAsync()
+    {
+        startDate = StartDate.GetValueOrDefault(DateTime.Now);
+        var url = $"sample-data/weather.json?date={startDate.ToString("yyyy-MM-dd")}";
+        using var req = new HttpRequestMessage(HttpMethod.Get, url);
+        req.SetBrowserResponseStreamingEnabled(true);
+
+        // send the request from the UI thread
+        var pr = Http.SendAsync(req, HttpCompletionOption.ResponseHeadersRead);
+        using var response = await pr;
```

Comment:
```suggestion
        using var response = await Http.SendAsync(req, HttpCompletionOption.ResponseHeadersRead);
```

Reply:
Also can I check: does the multithreaded runtime support performing HttpClient requests entirely on a background thread, without having to dispatch to the UI thread at all?

Reply:
I does it badly on current runtime main, but I have PR in progress and eventually, it will just work.